02/flight-service/src/main/java/cz/muni/fi/resources/FlightResource.java

```diff
@@ -0,0 +1,127 @@
+package cz.muni.fi.resources;
+
+import cz.muni.fi.model.Flight;
+import cz.muni.fi.service.FlightService;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.DELETE;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.PUT;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import org.jboss.resteasy.reactive.RestResponse;
+
+import java.util.List;
+
+/**
+ * This class is a REST resource that will be hosted on /flight
+ */
+@Path("/flight")
+public class FlightResource {
+
+    @Inject
+    FlightService flightService;
+
+
+    /**
+     * Get list of all flights
+     *
+     * @return list of all flights
+     */
+    @GET // This method process GET requests on /flight
+    @Produces(MediaType.APPLICATION_JSON) // This will set Content-Type header to application/json
+    public List<Flight> list() {
+        return flightService.listAll();
+    }
+
+    /**
+     * Create a new flight
+     *
+     * @param flight flight to create.
+     * @return created flight
+     */
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON) // This will indicate that this method consumes JSON
+    public RestResponse<Flight> create(Flight flight) { // Converts JSON from request body to Flight object
+        try {
+            var newFlight = flightService.createFlight(flight);
+            return RestResponse.status(Response.Status.CREATED, newFlight);
```

Comment:
this is kind of subjective, but maybe don't use RestResponse here but rather Response from jakarta ee. Just because we should teach the more broader API used, and not the proprietary ones if we don't have to.

Reply:
Ok I will create an issue to change it in all the lectures.